### PR TITLE
Feature/item template

### LINKED ISSUE #3: Item Template Component
The Item Template Component should characterize most or all items in the game.

### PR DESCRIPTION
**Pull request by ChrisKrane**
Add simple item template as a resource. 

- [x] Can add model as packed scene
- [x] Can add texture for displaying in inventory etc
- [x] Can add name and rarity
- [x] Can add Class specificity
- [ ] Can add item effects 

# Example model:
image

# Example inspector with data:
image

### CODE REVIEW

File: scripts/Items/Item.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+		get => _classSpecific;
+		set
+		{
+			_classSpecific = value;
+			NotifyPropertyListChanged();
+		}
+	}
+
+	[Export(PropertyHint.Enum, "Soldier,Melee,Mage,Engineer")]
+	public string ItemClass { set; get; }
```
Comment (CraZyB1336) [inline scripts/Items/Item.cs]: Perhaps a private set?

File: scripts/Items/Item.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+{
+	[ExportGroup("Item properties")]
+	[Export] public string itemName { set; get; }
+	[Export(PropertyHint.Enum, "Common,Rare,Legendary")]
+
+	public string itemGrade { set; get; }
+	
+	[ExportGroup("Item visuals")]
+	[Export] public Texture itemImage { set; get; }
+	[Export] public PackedScene itemModel { set; get; }
```
Comment (CraZyB1336) [inline scripts/Items/Item.cs]: Perhaps a private set? Since they are initialized in the inspector.

**Review by CraZyB1336 — APPROVED** [review]

**Review by CraZyB1336 — CHANGES_REQUESTED** [review]
Meant this button.
Look at the requested changes above

File: scripts/Items/Item.cs (hunk 1 above)
Comment (ChrisKrane) [reply scripts/Items/Item.cs]: Fixed

File: scripts/Items/Item.cs (hunk 2 above)
Comment (ChrisKrane) [reply scripts/Items/Item.cs]: fix fixed

**Review by CraZyB1336 — APPROVED** [review]
LGTM!

**Pull request merged**